editor/property_editor.h

```diff
@@ -207,6 +207,10 @@ class PropertyEditor : public Control {
 
 	bool updating_folding;
 
+	// Hacks to make collpse/expand works as expected.
+	bool tree_initialization;
+	bool tree_update_due_node_select;
```

Comment:
Both booleans are used for the exact same thing, so there's no need to have two of them. And then, it should likely be an optional bool parameter to `update_tree` instead of member variables that you update before and after calling it. Like `update_tree(bool p_init_folding = false)`, or any appropriate name that describes what this first branch does.